test/components/DayPickerRangeController_spec.jsx

```diff
@@ -1563,6 +1563,22 @@ describe('DayPickerRangeController', () => {
         expect(args.endDate.format()).to.equal(clickDate.clone().add(4, 'days').format());
       });
 
+      it('does not call props.onDatesChange with startDate === startDateOffset(date) and endDate === endDateOffset(date)', () => {
+        const clickDate = moment(today).clone().add(2, 'days');
+        const onDatesChangeStub = sinon.spy();
+        const wrapper = shallow((
+          <DayPickerRangeController
+            onDatesChange={onDatesChangeStub}
+            startDateOffset={day => day.subtract(2, 'days')}
+            endDateOffset={day => day.add(4, 'days')}
+            isOutsideRange={day => day.isAfter(moment(today))}
+          />
+        ));
+        wrapper.instance().onDayClick(clickDate);
+        /* eslint-disable-next-line no-unused-expressions */
+        expect(onDatesChangeStub.called).to.be.false;
```

Comment:
Please do not use no-op matchers; use `.to.equal(false)` instead.

Separately, `expect(stub).to.have,property(‘callCount’, 0)` has a much better failure message.